src/Curiosity/Runtime.hs

```diff
@@ -350,22 +344,19 @@ createUser runtime (newId, User.Signup {..}) = do
 createUserFull
   :: Runtime -> User.UserProfile -> STM (Either User.UserErr User.UserId)
 createUserFull runtime newProfile = do
-  mprofile <- selectUserById runtime newProfileId
+  mprofile <- liftA2 (<|>)
+                     (selectUserById runtime newProfileId)
+                     (selectUserByUsername runtime userName)
```

Comment:
I don't we can combine them together: we should have different error messages in different cases. Anyway, I think we'll have to rule-out identical IDs (the way we generate them should ensure that such an error doesn't happen), so I guess we can keep your change.